library/coretests/tests/option.rs

```diff
@@ -495,6 +495,25 @@ const fn option_const_mut() {
     */
 }
 
+/// Test that `Option::get_or_insert_*` are usable in const contexts, including with types that
+/// do not have `const Drop` (except for [`Option::get_or_insert()`], which is not const).
+#[test]
+fn option_const_get_or_insert() {
```

Comment:
Maybe split this test such that we can use a more unambiguous name don't have to mention an exception.

Reply:
Done.